packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -101,6 +101,107 @@ enum HeroFlightDirection {
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=Be9UH1kXFDw}
 ///
+/// {@tool dartpad --template=stateful_widget_scaffold_center}
+/// This sample shows a `Hero` with List item along with Icon: here we will be using a
+/// rectangle widget for icon. Icon widget in list item at first page , with Hero
+/// widget in list and Enlarged icon to display on Second Page.
+///
+/// Navigator.of(context).push() is needed to navigate to second page from
+/// first page using the SAME tag Making sure both widgets have same tag. Hero
+/// transitions use the same tag to identify the start and destination widgets.
+///
+/// ```dart
+///  Widget build(BuildContext context) {
+///    return HeroAnimationRecipe();
+///  }
+/// ```
+///
+/// ```dart preamble
+///  class HeroAnimationRecipe extends StatefulWidget {
+///
+///    @override
+///    _HeroAnimationRecipeState createState() => _HeroAnimationRecipeState();
+///  }
+///
+///  class _HeroAnimationRecipeState extends State<HeroAnimationRecipe> {
+///
+///    Widget _blueIconRectangle() {
+///     return Container(
+///       width: 50,
+///       height: 50,
+///       color: Colors.blue,
+///      );
+///    }
+///
+///    Widget _blueDetailRectangle() {
+///     return Container(
+///      width: 200,
+///       height: 200,
+///       color: Colors.blue,
+///     );
+///    }
```

Comment:
Could these methods (rectangle, detail) be reduced to just one that takes a size parameter?

Reply:
Thanks @Piinks